< build > add timer to module_pw

sunliang98 opened the pull request:
< fix >
fix the memory release bug in unit test (test2-2.cpp and test2-3.cpp).

sunliang98 committed:
< fix >
fix the memory release bug in unit test (test2-2.cpp and test2-3.cpp).

caic99 commented on source/module_pw/unittest/test2-2.cpp:
```diff
@@ -153,6 +154,9 @@ int main(int argc,char **argv)
+    MPI_Barrier(MPI_COMM_WORLD);     
+    delete [] rhog;
+    delete [] rhor;
+    if(tmp!=NULL) delete []tmp; 
```
Why not keeping `tmp` within `if(myrank == 0)` case? i.e. 
```c++
if(myrank==0) { 
complex<double> *tmp = new complex<double> [nx*ny*nz];
// do something
delete[] tmp;
}
```

Qianruipku committed:
< prof >
	Modify some places to make the program faster.
	cancel the assert(n!=0) in global_function.h
	change 2d/2d_r2c/2d_c2r to 1d + 1d. The latter one is faster.
	change out-of-place 2d to in-place.
	in the future, we may change out-of-place r2c/c2r to in-place

Qianruipku committed:
< prof >
	Rewrite gatterscatter function.
    Now it has higher parallel efficiency.

Qianruipku committed:
change executefftw function to a series of functions

Qianruipku committed:
< prof >
	change in-place fft to out-of-place fft for fftz

Qianruipku committed:
change istot2ixy to istot2bigixy

Qianruipku committed:
add liy,riy

Qianruipku committed:
< prof >
	fft in x-direction are reduced by half

Qianruipku committed:
< prof >
	change out-of-place r2c to in-place r2c
	It is faster

Qianruipku committed:
< prof >
	add riy and liy for gamma_only, thus half time is reduce for y fft
	change rix/lix to riy/liy

sunliang98 committed:
Merge branch 'planewave' of https://github.com/Qianruipku/abacus-develop into planewave

Qianruipku committed:
< add >
	add float data for pw module
	add test for float

sunliang98 committed:
add timer to module_pw

sunliang98 committed:
Merge branch 'planewave' of https://github.com/Qianruipku/abacus-develop into planewave
